notebooks/working/ho_deconfounder_on_sim_data.py

```diff
@@ -0,0 +1,512 @@
+
+mnl_specification["total_travel_distance"] = [1, 2, 3, 7, 8]
+mnl_names["total_travel_distance"] = ['Travel Distance, units:mi (Drive Alone)',
+                                      'Travel Distance, units:mi (SharedRide-2)',
+                                      'Travel Distance, units:mi (SharedRide-3+)',
+                                      'Travel Distance, units:mi (Walk)',
+                                      'Travel Distance, units:mi (Bike)']
+
+# mnl_specification["cross_bay"] = [[2, 3], [4, 5, 6]]
+# mnl_names["cross_bay"] = ["Cross-Bay Tour (Shared Ride 2 & 3+)",
+#                           "Cross-Bay Tour (All Transit Modes)"]
+mnl_specification["cross_bay"] = [[2, 3]]
+mnl_names["cross_bay"] = ["Cross-Bay Tour (Shared Ride 2 & 3+)"]
+
+mnl_specification["household_size"] = [[2, 3]]
+mnl_names["household_size"] = ['Household Size (Shared Ride 2 & 3+)']
+
+mnl_specification["num_kids"] = [[2, 3]]
+mnl_names["num_kids"] = ["Number of Kids in Household (Shared Ride 2 & 3+)"]
+# -
+
+# # Deconfounder
+
+# ## Method 1
+#
+# In this method, we used all the variables in each mode's utility specification to recover its confounder, in line with Blei et al.'s approach in their paper.
+
+# +
+## Return the variables of each utility specification
+
+spec_dic = specifications(mnl_specification=mnl_specification, num_modes=8)
+spec_dic
+# -
+
+confounder_vectors = []
+holdout_dfs = []
+masks_df = []
+rows_df =[]
+latent_dim = 1
+for i in data['mode_id'].unique():
+
+    data_mode_i = data[data['mode_id']==i]
+    # standardize the data for PPCA
+    print("Analysis for mode: ", i)
+    print("-------------------------------------------------------------------------------------------")
+    X_columns = spec_dic[i]
+    
+    if 'intercept' in X_columns:
+        X_columns.remove('intercept')
+        
+    if 'total_travel_distance' in X_columns:
+        X_columns.remove('total_travel_distance')
+        
+    print(i, X_columns)
+    
+    X = np.array((data_mode_i[X_columns] - data_mode_i[X_columns].mean())/data_mode_i[X_columns].std())
+    
+    confounders, holdouts, holdoutmasks, holdoutrow= confounder_ppca(holdout_portion=0.2, X=X, latent_dim=latent_dim)
+
+    confounder_vectors.append(confounders)
+    holdout_dfs.append(holdouts)
+    masks_df.append(holdoutmasks)
+    rows_df.append(holdoutrow)
+
+# ## Method 2
+#
+# In this method, we only run the deconfounder's factor model on travel_time and travel_cost, which are the only variables confounded with travel_distance in our simulation
+
+# +
+xs = ['total_travel_time', 'total_travel_cost']
+X_DA = (data[data['mode_id']==1][xs] - data[data['mode_id']==1][xs].mean())/data[data['mode_id']==1][xs].std()
+
+X_SR2 = (data[data['mode_id']==2][xs] - data[data['mode_id']==2][xs].mean())/data[data['mode_id']==2][xs].std()
+
+X_SR3 = (data[data['mode_id']==3][xs] - data[data['mode_id']==3][xs].mean())/data[data['mode_id']==3][xs].std()
```

Comment:
Instead of copying the same logic three times, I'd suggest creating a function `get_standardized_data` that takes as input `df, column_list` and returns the standardized dataframe.